src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -274,13 +280,17 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
                 // if we can't allocate it on a node, ignore it, for example, this handles
                 // cases for only allocating a replica after a primary
                 Decision decision = allocation.deciders().canAllocate(shard, node, allocation);
+                if (decision.type() != Decision.Type.THROTTLE) {
```

Comment:
I'm a bit confused by this. If `canBeAllocatedToAtLeastOneNode` is false,  we already skip this shard. If `canBeAllocatedToAtLeastOneNode` is true, it means that at least one decision was YES, which also implies not THROTTLE, meaning that allAreThrottled will also be false. This makes allAreThrottled redundant? or am I missing something?